crates/blockchain/blockchain.rs

```diff
@@ -143,17 +143,13 @@ pub fn validate_block(
     let spec = spec_id(state.database(), block.header.timestamp).unwrap();
 
     // Verify initial header validity against parent
-    let mut valid_header = validate_block_header(&block.header, parent_header);
+    validate_block_header(&block.header, parent_header).map_err(InvalidBlockError::from)?;
 
-    valid_header = match spec {
-        SpecId::CANCUN => {
-            valid_header && validate_cancun_header_fields(&block.header, parent_header)
-        }
-        _ => valid_header && validate_no_cancun_header_fields(&block.header),
+    match spec {
```

Comment:
Can we make this match more explicit?
Since this is error handling, maybe we could match with the explicit variants? I've noticed the SpecId is represented as u8, so maybe we can take advantage of that?

Reply:
There are lots of them, opted for renaming the `_` [29352ce](https://github.com/lambdaclass/ethereum_rust/pull/699/commits/29352ce3e3390b1882bc42b84e2a397ee46329b3)